plone/app/users/browser/register.py

```diff
@@ -60,9 +72,9 @@ class BaseRegistrationForm(AutoExtensibleForm, form.Form):
     # this attribute indicates if user was successfully registered
     _finishedRegister = False
 
-    def __init__(self, *args, **kwargs):
-        super(BaseRegistrationForm, self).__init__(*args, **kwargs)
-        self.schema = getUtility(IRegisterSchemaProvider).getSchema()
+    @property
```

Comment:
I have implemented a simple cache solution based on your last changes here https://github.com/plone/plone.app.users/pull/47
I am just waiting for Jenkins to be up again to see if tests are passing.